lib/src/ast_extensions.dart

```diff
@@ -149,58 +151,68 @@ extension ExpressionExtensions on Expression {
   ///
   /// Completely empty delimited constructs like `[]` and `foo()` don't allow
   /// splitting inside them, so are not considered block-like.
-  bool get canBlockSplit {
+  bool get canBlockSplit => blockFormatType != BlockFormat.none;
+
+  /// When this expression is in an argument list, what kind of block formatting
+  /// category it belongs to.
+  BlockFormat get blockFormatType {
     // Unwrap named expressions to get the real expression inside.
-    var expression = this;
-    if (expression is NamedExpression) {
-      expression = expression.expression;
+    if (this case NamedExpression named) {
```

Comment:
[nit] Can we put this in the `switch` cases?

Reply:
Good catch, yes.